src/main/java/org/vafer/jdependency/Clazzpath.java

```diff
@@ -123,31 +123,25 @@ InputStream getInputStream() throws IOException {
     }
 
     public ClazzpathUnit addClazzpathUnit( final InputStream pInputStream, final String pId ) throws IOException {
-
-        final JarInputStream inputStream = new JarInputStream(pInputStream);
-
-        try {
-
+        try (final JarInputStream inputStream = new JarInputStream(pInputStream)){
             Iterable<Resource> resources = asStream(inputStream)
-                .map(e -> e.getName())
-                .filter(name -> isValidResourceName(name))
+                .map(JarEntry::getName)
+                .filter(Clazzpath::isValidResourceName)
                 .map(name -> (Resource) new Resource(name) {
-                    InputStream getInputStream() throws IOException {
+                    @Override
+					InputStream getInputStream() throws IOException {
                         return inputStream;
                     }
                 })::iterator;
 
            return addClazzpathUnit(resources, pId, false);
-
-        } finally {
-            inputStream.close();
```

Comment:
I am not aware that `try()` will implicitly call `close()` - does it?

Reply:
Actually it does. That is the why it Closeable now extends AutoCloseable.
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
The importend part are the round brackets

Reply:
Interesting. Wasn't aware of that. Sneaky.